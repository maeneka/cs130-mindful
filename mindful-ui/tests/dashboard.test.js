import React from 'react'
import Dashboard from '../src/components/dashboard';
import { h } from 'preact';
import renderer from 'react-test-renderer';
// See: https://github.com/preactjs/enzyme-adapter-preact-pure
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Test of the Dashboard UI', () => {
	test('Dashbaord UI for all front end elements', () => {
		const wrapper = shallow(<Dashboard />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	test('Dashboard UI main text displays', () => {
		const wrapper = shallow(<Dashboard />);
		expect(wrapper.find('h3').text()).toBe('mindful dashboard');
	});

	test('Dashboard UI back to settings', () => {
		const wrapper = shallow(<Dashboard />);
		expect(wrapper.find('a').find('p').text()).toBe('SETTINGS');
	});

});
