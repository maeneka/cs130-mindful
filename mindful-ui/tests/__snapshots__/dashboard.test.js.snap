// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of the Dashboard UI Dashbaord UI for all front end elements 1`] = `
<div
  className="page"
>
  <div>
    <h3>
      mindful dashboard
    </h3>
    <div
      className="nav"
    >
      <a
        className="mini-button"
        href="#"
        onClick={[Function]}
      >
        TODAY
      </a>
      <a
        className="white-btn"
        href="#"
        onClick={[Function]}
      >
        WEEK
      </a>
      <a
        className="white-btn"
        href="#"
        onClick={[Function]}
      >
        MONTH
      </a>
    </div>
    <h1
      className="hr"
    >
      5 Hrs
    </h1>
    <h2
      className="min"
    >
      43 Minutes
    </h2>
    <Delta
      delta={-5}
      timescale="yesterday"
    />
  </div>
  <div>
    <p
      className="gray"
    >
      BREAKDOWN
    </p>
    <div
      className="breakdown"
    >
      <Metric
        time={1.5}
        url="facebook.com"
        width={100}
      />
      <Metric
        time={1.5}
        url="facebook.com"
        width={100}
      />
      <Metric
        time={1.5}
        url="facebook.com"
        width={100}
      />
      <Metric
        time={1.5}
        url="facebook.com"
        width={100}
      />
    </div>
  </div>
  <a
    className="small-button"
    href="#"
    onClick={[Function]}
  >
    <p>
      SETTINGS
    </p>
  </a>
</div>
`;
